src/app/project/project-permissions/permissions-table-rows/permissions-table-rows.component.ts

```diff
@@ -39,36 +46,100 @@ export class PermissionsTableRowsComponent implements OnInit, TableComponent {
    *
    * @return {void}
    */
-  ngOnInit() {
+  ngOnInit(): void {
     this.currentProject = this.storageService.state.currentProject.data;
     this.entries = this.data.data;
+    this.paginationData = this.data.paginationData;
+    this.route.params
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(params => {
+        this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params);
+      });
   }
 
   /**
    * Opens a new snack bar notification message with a duration of 2 seconds, and executes an action.
    *
    * @param {string} message A snack bar notification message.
    * @param {string} action A snack bar notification action.
-   * @returns {void}
+   * @return {void}
    */
    public openSnackBar(message: string, action: string): void {
     this.snackBar.open(message, action, {
       duration: 2000,
     });
   }
 
-  hasProjectPermission(user) {
+  /**
+   * Determines if a user has project permission.
+   *
+   * @param {User} user The user to check.
+   * @returns {boolean}
+   */
+  hasProjectPermission(user: User): boolean {
     return user.projectPermissions.includes(this.currentProject._id);
   }
 
+  /**
+   * Removes duplicate user entries.
+   *
+   * @param {User[]} returnedUsers The user list that will be checked for duplicate users.
+   * @returns {User[]}
+   */
+  removeDuplicateUsers(returnedUsers: User[]): User[] {
+    let userNames = [];
+    let validatedUsers = [];
+    if (Array.isArray(this.entries)) {
+    returnedUsers.forEach(user => {
+      if (!userNames.includes(user.displayName)) {
+        // If no duplicate name is found, validate the user
+        userNames.push(user.displayName);
+        validatedUsers.push(user);
+      } else if (user.projectPermissions?.includes(this.currentProject._id)) { 
+        // If a duplicate user is found and they have project permission, replace matching entry if it doesn't have permissions
+        const matchingUserIndex = validatedUsers.findIndex((usr) => usr.displayName === user.displayName);
+        validatedUsers[matchingUserIndex] = validatedUsers[matchingUserIndex]?.projectPermissions?.includes(this.currentProject._id) ? validatedUsers[matchingUserIndex] : user;
```

Comment:
Does this line remove duplicates that both contain project permissions? I can see above that you removed duplicates by displayName

Reply:
Right so if there is no duplicate found it will add the name to the names array and add the entry to validated entries. Otherwise, if it does find a duplicate entry but the new one has permissions, it checks if the old one had permissions, and if it didn't, it replaces it with the new one. This insures that if someone has permissions, it will always show up.